great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -136,11 +136,14 @@ def _check_xor_input_args(cls, values: dict) -> dict:
         connection_string: str | ConnectionDetails | None = values.get("connection_string")
         if connection_string:
             # Method 1 - connection string
-            if isinstance(connection_string, (str, ConfigStr)):
-                return values
-            # Method 2 - individual args (account, user, and password are bare minimum)
-            elif isinstance(connection_string, ConnectionDetails) and bool(
-                connection_string.account and connection_string.user and connection_string.password
+            if (
+                isinstance(connection_string, (str, ConfigStr))
+                or isinstance(connection_string, ConnectionDetails)
+                and bool(
+                    connection_string.account
+                    and connection_string.user
+                    and connection_string.password
+                )
```

Comment:
This change gives me pause.
Even if it's more performant the old method is more readable.
I don't feel strongly though.